### ISSUE TITLE
After upgrade to Fedora 37 - "global flags not at the start of the expression" with any option

### ISSUE THREAD

**Issue by Hui-Lui** [yt-dlp/yt-dlp#6514]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2023.03.04** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

Since I upgraded to Fedora 37 (end of February), I always get the "global flags not at the start" error.
Today I installed the newest version via the package manager dnf, 2023.03.04-1.fc37, but the problem still persists.

I also want to mention, that in the midst of February, while I still had Fedora 36, after one update of yt-dlp, it did not download any youtube video, so I checked the version via `yt-dlp --version`. 
It showed me some version beginning with 2021, don't know which exactly anymore, although the package version (listed by dnf) was something with 2023. 
I found that odd, but ignored it for sometime, because for other websites it still worked.

Then after the upgrade from Fedora 36 to 37 I get the mentioned error on any option while executing yt-dlp.
So I can't even look up the current version via `yt-dlp --version`.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
yt-dlp -vU --version
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/usr/local/bin/yt-dlp/__main__.py", line 16, in <module>
  File "/usr/local/bin/yt-dlp/yt_dlp/__init__.py", line 16, in <module>
  File "/usr/local/bin/yt-dlp/yt_dlp/options.py", line 26, in <module>
  File "/usr/local/bin/yt-dlp/yt_dlp/downloader/__init__.py", line 36, in <module>
  File "/usr/local/bin/yt-dlp/yt_dlp/downloader/youtube_live_chat.py", line 14, in <module>
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/youtube.py", line 3012, in <module>
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/youtube.py", line 4125, in YoutubeTabIE
  File "/usr/lib64/python3.11/re/__init__.py", line 227, in compile
    return _compile(pattern, flags)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/__init__.py", line 294, in _compile
    p = _compiler.compile(pattern, flags)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/_compiler.py", line 743, in compile
    p = _parser.parse(p, flags)
        ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/_parser.py", line 980, in parse
    p = _parse_sub(source, state, flags & SRE_FLAG_VERBOSE, 0)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/_parser.py", line 455, in _parse_sub
    itemsappend(_parse(source, state, verbose, nested + 1,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/_parser.py", line 863, in _parse
    p = _parse_sub(source, state, sub_verbose, nested + 1)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/_parser.py", line 455, in _parse_sub
    itemsappend(_parse(source, state, verbose, nested + 1,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/_parser.py", line 841, in _parse
    raise source.error('global flags not at the start '
re.error: global flags not at the start of the expression at position 8 (line 1, column 9)
```

**Comment by Grub4K** [comment]
Maybe your PATH contains another yt-dlp version thats outdated? Try uninstalling using you package manager, then runing `which yt-dlp` to see if any other versions may be present that get run instead. If so, clean them up and try again after a reinstall.

If it was indeed setup correctly, try using the [yt-dlp](https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp) binary directly and see if that works. If that errors then post the verbose log of that version and use [yt-dlp_linux](https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp_linux) for now as a workaround.

**Comment by Hui-Lui** [comment]
Thank you, there does seem to be indeed an older version.
`which yt-dlp`
gave me:
`/usr/local/bin/yt-dlp`
and `whereis yt-dlp` gave me the same output.
So for cleaning up, does it suffice to just delete that file, before reinstalling yt-dlp via package manager?

**Comment by pukkandan** [comment]
Depends on how you installed it. If you downloaded it urself, deleting works. But if it was created by a PM, use it to properly uninstall.

**Comment by Hui-Lui** [comment]
I searched my `.bash_history `file and the first entry with yt-dlp gives me: `sudo dnf install yt-dlp`. 
So it was indeed installed by the package manager.

I already removed yt-dlp by the package manager and `dnf list installed yt-dlp` gives me:
`Error: No matching Packages to list`

But `which yt-dlp` and `whereis yt-dlp` still gives me:
`/usr/local/bin/yt-dlp`

----
Ok, I just used catfish file search to locate other yt-dlp files and found:
`yt-dlp` and `yt-dlp.tar.gz` in my downloads folder. Those and `/usr/local/bin/yt-dlp` have a timestamp from 2021, so I guess, I could delete them.
